src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs

```diff
@@ -825,25 +834,95 @@ private static LiftingResult UserDefinedBinaryOperatorCanBeLifted(TypeSymbol lef
                     {
                         continue;
                     }
+
                     var better = BetterOperator(candidates[i].Signature, candidates[j].Signature, left, right, ref useSiteDiagnostics);
-                    if (better == BetterResult.Left)
+                    if (better == BetterResult.Left && candidates[j].Kind != OperatorAnalysisResultKind.Worse)
```

Comment:
I was hoping that we can *save* some CPU cycles here, since we already did some checks earlier. Wouldn't this change make it slightly worse then, or we're just optimizing the success path?